packages/block-library/src/video/edit.js

```diff
@@ -161,7 +161,7 @@ function VideoEdit( {
 			const embedBlock = createUpgradedEmbedBlock( {
 				attributes: { url: newSrc },
 			} );
-			if ( undefined !== embedBlock ) {
+			if ( undefined !== embedBlock && onReplace ) {
```

Comment:
Do you think it would be better to check explicitly if it's locked?

Reply:
Currently, `onReplace` is only undefined when a block cannot be removed, but there might be more conditions in the future. I think it makes more sense to check if replace callback is available before calling it.

BlockEdito condition: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/block.js#L148-L149

Reply:
Sounds good, thanks!